tests/test_schemas.py

```diff
@@ -1,294 +1,645 @@
+    @pytest.mark.parametrize("node_name", ['', 'Diagnosis'], ids=['empty_node_name', "Diagnosis_node_name"])
+    @pytest.mark.parametrize("blank", [True, False], ids=["True_blank", "False_blank"])
+    def test_get_range_schema(self, helpers, data_model, node_range, node_name, blank):
+        dmjs = helpers.get_data_model_json_schema(data_model_name=data_model)
+        range_schema = dmjs.get_range_schema(node_range=node_range, node_name=node_name, blank=blank)
+
+        # check node_name is recoreded as the key to the array schema
+        assert node_name in range_schema
+
+        # Check that blank value is added at the end of node_range, if true
+        if blank:
+            assert range_schema[node_name]['enum'][-1]== ''
+            assert len(range_schema[node_name]['enum'])==len(node_range)+1
+        else:
+            assert range_schema[node_name]['enum']== node_range
+            assert len(range_schema[node_name]['enum'])==len(node_range)
+
+    @pytest.mark.parametrize("data_model", ['example.model.csv', 'example.model.jsonld'], ids=["csv", "jsonld"])
+    @pytest.mark.parametrize("source_node", ['', 'Patient'], ids=['empty_node_name', "patient_source"])
+    @pytest.mark.parametrize("schema_name", ['', 'Test_Schema_Name'], ids=['empty_schema_name', "schema_name"])
+    def test_get_json_validation_schema(self, helpers, data_model, source_node, schema_name):
+        dmjs = helpers.get_data_model_json_schema(data_model_name=data_model)
+
+        try:
+            # Get validation schema
+            json_validation_schema = dmjs.get_json_validation_schema(source_node=source_node, schema_name=schema_name)
+
+            # Check Keys in Schema
+            expected_jvs_keys = ['$schema', '$id', 'title', 'type', 'properties', 'required', 'allOf']
+            actual_jvs_keys = list( json_validation_schema.keys())
+            assert expected_jvs_keys == actual_jvs_keys
+
+            # Check title
+            assert schema_name == json_validation_schema['title']
+
+            # Check contents of validation schema
+            assert 'Diagnosis' in json_validation_schema['properties']
+            assert 'Cancer' in json_validation_schema['properties']['Diagnosis']['enum']
+        except:
+            # Should only fail if no source node is provided.
+            assert source_node == ''
+
+class TestDataModelJsonLd:
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    def test_init(self, helpers, data_model):
+        # Get Graph
+        graph_data_model = generate_graph_data_model(helpers, data_model_name=data_model)
+
+        # Instantiate DataModelJsonLD
+        data_model_jsonld = DataModelJsonLD(Graph=graph_data_model)
+
+        # Test that __init__ is being set up properly 
+        assert type(data_model_jsonld.graph) == nx.MultiDiGraph
+        assert type(data_model_jsonld.rel_dict) == dict
+        assert 'required' in data_model_jsonld.rel_dict
```

Comment:
why do we need to check if 'required' is in data_model_jsonld.rel_dict ?
--Migrated from: https://github.com/Sage-Bionetworks/schematic/pull/1313